### PR TITLE
Check keys on load

### PR DESCRIPTION
**Pull request by swhite2401**
Instead of printing a message without explanation when loading a matfile with the wrong key, an AtError is raised that prints the list of available keys

### CODE REVIEW

**Comment by swhite2401** [comment]
Oups, I checked out this branch from another dev branch, please ignore the changes in orbit.py

File: pyat/at/load/allfiles.py [hunk 1]
```diff
@@ -38,10 +38,7 @@ def load_lattice(filepath, **kwargs):
     Known extensions are:
     """
     _, ext = os.path.splitext(filepath)
-    try:
-        return _load_extension[ext.lower()](filepath, **kwargs)
-    except KeyError:
```
Comment (lfarv) [inline pyat/at/load/allfiles.py:43]: Do you think that the `KeyError` message is clear enough ? If so, you can apply the same simplification to `save_lattice`.

**Comment by swhite2401** [comment]
Ah ok now I understand. This was very confusing as it was also catching the `KeyError` coming from the unknown `mat_key` in `matfile.py`. Now this is handled and throws an `AtError` so it should be ok. I put back the message (a bit different).

**Review by lfarv — APPROVED** [review]
Very good !

**Pull request merged**